update config flow for ecobee (#45087)
home-assistant/home-assistant.github.io: source/_integrations/ecobee.markdown

The **ecobee** {% term integration %} lets you control and view sensor data from [ecobee](https://ecobee.com) thermostats.

<p class='img'>
  <img src='/images/screenshots/ecobee-sensor-badges.png' />
  <br />
  <img src='/images/screenshots/ecobee-thermostat-card.png' />
</p>

{% note %}

> Since March 2026, it is no longer required to get a [developer API key](https://www.ecobee.com/developers/) to use this integration. Existing API keys will continue to function. If you revoke your existing key in the ecobee portal, you will need to remove the `ecobee` configuration in the Integrations panel, and then configure the integration again.

Reviewer: what do you mean by: "remove the `ecobee` configuration in the Integrations panel"
Go to {% my integrations title="**Settings** > **Devices & services**" %} and remove the **Ecobee** integration, and then install it again?

Author: clarified

{% endnote %}

> 

Reviewer: Would it be worth adding a prerequisite section, stating that they need an ecobee account and a thermostat registered on ecobee's website? Or is this no longer required?

Author: Definitely necessary :) Added this section

{% include integrations/config_flow.md %}

{% configuration_basic %}
API key:

>   description: If you have developer API key, use this field and ignore the others. **If you are logging in without an API key, leave this field blank, and use the others.**

Reviewer: ```suggestion
  description: If you have a developer API key, use this field and ignore the others. If you are logging in without an API key, leave this field blank and use username and password.
```
We use [bold to markup UI strings](https://developers.home-assistant.io/docs/documenting/general-style-guide), not to highlight text.

The **ecobee** {% term integration %} lets you control and view sensor data from [ecobee](https://ecobee.com) thermostats.

<p class='img'>
  <img src='/images/screenshots/ecobee-sensor-badges.png' />
  <br />
  <img src='/images/screenshots/ecobee-thermostat-card.png' />
</p>

## Prerequisites

> To set up the connection, you will need to have a username and password for [ecobee.com](https://ecobee.com). Set this up before adding the integration. You can add devices either before or after you configure the service in Home Assistant, but having them connected to your ecobee.com account ahead of time is recommended to confirm that they are picked up by the service correctly.

Reviewer: ```suggestion

- Username and password for [ecobee.com](https://ecobee.com). 
   - You will need it when adding the integration to set up a connection between the integration and Home Assistant. 
- Have the devices connected to your ecobee.com account.
   - You can add devices either before or after you configure the service in Home Assistant, but having them connected to your ecobee.com account ahead of time is recommended to confirm that they are picked up by the service correctly.
```

{% note %}

> Since version 2026.3, it is no longer required to get a [developer API key](https://www.ecobee.com/developers/) to use this integration. Existing API keys will continue to function. If you revoke your existing key in the ecobee portal, the integraion will fail, and you will need to remove the service in Home Assistant and set it up again.

Reviewer: ```suggestion
Since version 2026.3, it is no longer required to get a [developer API key](https://www.ecobee.com/developers/) to use this integration. Existing API keys will continue to function. If you revoke your existing key in the ecobee portal, the integration will fail, and you will need to remove the service in Home Assistant and set it up again.
```
